cilium/cmd/encrypt_status.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os/exec"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/cilium/cilium/api/v1/client/daemon"
+	"github.com/cilium/cilium/api/v1/models"
+	"github.com/cilium/cilium/pkg/command"
+	"github.com/cilium/cilium/pkg/common"
+	"github.com/prometheus/procfs"
+	"github.com/vishvananda/netlink"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// Cilium uses reqid 1 to tie the IPsec security policies to the matching state’
+	CiliumReqId = "1"
+)
+
+type void struct{}
+
+var (
+	voidType    void
+	countErrors int
+)
+
+var encryptStatusCmd = &cobra.Command{
+	Use:   "status",
+	Short: "Display the current encryption state",
+	Run: func(cmd *cobra.Command, args []string) {
+		common.RequireRootPrivilege("cilium encrypt status")
+		getEncryptionMode()
+	},
+}
+
+func init() {
+	encryptCmd.AddCommand(encryptStatusCmd)
+	command.AddJSONOutput(encryptStatusCmd)
+}
+
+func getXfrmStats() (int, map[string]int) {
+	fs, err := procfs.NewDefaultFS()
+	if err != nil {
+		Fatalf("Cannot get a new proc FS: %s", err)
+	}
+	stats, err := fs.NewXfrmStat()
+	if err != nil {
+		Fatalf("Failed to read xfrm statistics: %s", err)
+	}
+	v := reflect.ValueOf(stats)
+	errorMap := make(map[string]int)
+	for i := 0; i < v.NumField(); i++ {
```

Comment:
Note that `NumField` will panic if `v` is not a struct. I think it is OK in this case but it would be best to check the type with `v.Type().Kind()` to be `reflect.Struct`.